detection-rules/link_cloudflare_service_abuse.yml

```
@@ -0,0 +1,73 @@
+name: "Suspicious Links to Cloudflare R2 and Edge Services"
+description: "Detects links to Cloudflare R2 storage buckets, Pages, and Workers domains from unsolicited or previously malicious senders who are not on a trusted sender list or have failed DMARC authentication."
+type: "rule"
+severity: "medium"
+source: |
+  type.inbound
+  and any(body.links,
+          // R2 buckets with pub-{32hex}.r2.dev pattern
+          regex.icontains(.href_url.url, 'https://pub-[a-f0-9]{32}\.r2\.dev')
+          // generic r2.dev domains
+          or strings.icontains(.href_url.domain.root_domain, "r2.dev")
+          // Cloudflare Pages
+          or strings.icontains(.href_url.domain.root_domain, "pages.dev")
+          // Cloudflare Workers
+          or strings.icontains(.href_url.domain.root_domain, "workers.dev")
+  )
```

```suggestion
  type.inbound
  and any(body.links,
          // Check root domain first 
          .href_url.domain.root_domain in ("r2.dev", "pages.dev", "workers.dev")
          // Check the specific pub-{hex}.r2.dev subdomain pattern only for r2.dev
          or (
              .href_url.domain.root_domain == "r2.dev"
              and regex.icontains(.href_url.domain.subdomain, '^pub-[a-f0-9]{32}$')
          )
  )
```
You can combine the list of these root domains into an "in" statement for an optimization bump. Also makes it easier to add new domains in the future!